File: c#/trabajo4.cs
using System;

class Program
{
    static void Main()
    {
        int[,] mat = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };

        // Recorrer por renglones
        Console.WriteLine("Recorrer por renglones:");
        for (int i = 0; i < mat.GetLength(0); i++) // filas
        {
            for (int j = 0; j < mat.GetLength(1); j++) // columnas
            {
                Console.Write(mat[i, j] + " ");
            }
            Console.WriteLine(); // salto de línea al terminar la fila
        }

        // Recorrer por columnas
        Console.WriteLine("Recorrer por columnas:");
        for (int j = 0; j < mat.GetLength(1); j++) // columnas
        {
            for (int i = 0; i < mat.GetLength(0); i++) // filas
            {
                Console.Write(mat[i, j] + " ");
            }
            Console.WriteLine(); // salto de línea al terminar la columna
        }
    }
}

File: java/trabajo4.java
public class MatrizRecorrido {
    public static void main(String[] args) {
        int[][] mat = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };

        // Recorrer por renglones
        System.out.println("Recorrer por renglones:");
        for (int i = 0; i < mat.length; i++) { // filas
            for (int j = 0; j < mat[i].length; j++) { // columnas
                System.out.print(mat[i][j] + " ");
            }
            System.out.println(); // salto de línea al terminar la fila
        }

        // Recorrer por columnas
        System.out.println("Recorrer por columnas:");
        for (int j = 0; j < mat[0].length; j++) { // columnas
            for (int i = 0; i < mat.length; i++) { // filas
                System.out.print(mat[i][j] + " ");
            }
            System.out.println(); // salto de línea al terminar la columna
        }
    }
}
